.github/workflows/kubectl-az.yml

```diff
@@ -0,0 +1,199 @@
+name: Microsoft Azure CLI kubectl plugin CI
+env:
+  GO_VERSION: 1.17 # TODO: Update
+  AZURE_PREFIX: kubectl-az-ci
+concurrency:
+  # Only one workflow can run at a time unless
+  # we create a new AKS cluster per github_ref (branch)
+  group: kubectl-az-ci
+
+on:
+  pull_request:
+    branches:
+      - main
```

Comment:
Nit: Maybe remove the `branches: -main` here. So the CI will run when we have a PR for another branch than `main`

Reply:
Umm. You mean if we create a PR from `foo` branch to `bar` branch? My idea was to only run CI when someone creates a PR to the `main` branch

Reply:
Yep. Agree with Burak, it might happen that we want to merge into a branch different than `main`.

Reply:
In addition, somethings one start working on a PR based on another PR. In that case, the destination branch is not the `main`, but we still want the CI to run.

Reply:
Makes sense. That happened faster then expected. I updated the PR now! :)